### PR TITLE
QUARKUS-1676 Add tests for 'MAIF - mTLS with OIDC Server'

### PR DESCRIPTION
**Pull request by michalvavrik**
This commit add tests for a mutual TLS with OIDC as suggested by [a test plan](https://github.com/quarkus-qe/quarkus-test-plans/pull/76). There are tests for JKS/PKCS12 keystore types, a fail test when a keystore file is completely missing, test with Keycloak and RH SSO 7.3/7.5 in role of a OIDC server and test with keystore file whose file type can't be detected by a file extension.

### CODE REVIEW

File: security/oidc-client-mutual-tls/src/test/java/io/quarkus/ts/security/oidcclient/mtls/AuthenticationTestSuite.java
Comment (pjgg) [inline security/oidc-client-mutual-tls/src/test/java/io/quarkus/ts/security/oidcclient/mtls/AuthenticationTestSuite.java:28]: This commend should be reviewed

File: security/oidc-client-mutual-tls/src/test/java/io/quarkus/ts/security/oidcclient/mtls/AuthenticationTestSuite.java
Comment (pjgg) [inline security/oidc-client-mutual-tls/src/test/java/io/quarkus/ts/security/oidcclient/mtls/AuthenticationTestSuite.java:9]: Could be renamed this interface to something more "meaningful", something like `KeycloakMtlsAuthN`, because all of these names are going to be used by "Polarion" in order to generate a report. 

So if for example `verifyAuthenticated` fails, we are going to have an alert like: 
`KeycloakMtlsAuthN.verifyAuthenticated` fails.

File: security/oidc-client-mutual-tls/src/test/java/io/quarkus/ts/security/oidcclient/mtls/MutualTlsKeycloakService.java
Comment (pjgg) [inline security/oidc-client-mutual-tls/src/test/java/io/quarkus/ts/security/oidcclient/mtls/MutualTlsKeycloakService.java:10]: This implementation should be merged into QuarkusTestFramework [keycloakService](https://github.com/quarkus-qe/quarkus-test-framework/blob/main/quarkus-test-service-keycloak/src/main/java/io/quarkus/test/bootstrap/KeycloakService.java). Could you create an issue in [QuarkusTestFramework](https://github.com/quarkus-qe/quarkus-test-framework) in order to add this issue to the backlog.

File: security/oidc-client-mutual-tls/src/test/java/io/quarkus/ts/security/oidcclient/mtls/AuthenticationTestSuite.java
Comment (pjgg) [inline security/oidc-client-mutual-tls/src/test/java/io/quarkus/ts/security/oidcclient/mtls/AuthenticationTestSuite.java:9]: I am not sure if this interface should be a class as `IncorrectKsFileTypeOidcMtlsIT`
Maybe could be a class and extends `BaseOidcMtlsIT`

File: security/oidc-client-mutual-tls/src/test/java/io/quarkus/ts/security/oidcclient/mtls/MutualTlsKeycloakService.java
Comment (pjgg) [inline security/oidc-client-mutual-tls/src/test/java/io/quarkus/ts/security/oidcclient/mtls/MutualTlsKeycloakService.java:64]: typo: passthrough -> passThrough

File: security/oidc-client-mutual-tls/src/test/resources/openshift-rh-sso-deployment-template.yml
Comment (pjgg) [inline security/oidc-client-mutual-tls/src/test/resources/openshift-rh-sso-deployment-template.yml:2]: I think that would be great to add this template to [Quarkus-test-framework](https://github.com/quarkus-qe/quarkus-test-framework) so based on `KeycloakService` configuration the framework is able to deploy keycloak with Mtls or a regular one. Please add this feature as part of the previously created issue

**Review by pjgg — CHANGES_REQUESTED** [review]
Missing add a reference in project `README.md`, that explains what this module does. You could use as an example some of the previous modules ex: `security/basic`

Some of the `test/resources` certificates already exist in `main/resource` folder, so you don't need to duplicate it, just use `main/resource` certificates.

File: security/oidc-client-mutual-tls/src/test/java/io/quarkus/ts/security/oidcclient/mtls/AuthenticationTestSuite.java
Comment (michalvavrik) [reply security/oidc-client-mutual-tls/src/test/java/io/quarkus/ts/security/oidcclient/mtls/AuthenticationTestSuite.java]: I removed the comment and added a new test. Now I think context (method name in context of its class) shall clarify method itself.

File: security/oidc-client-mutual-tls/src/test/java/io/quarkus/ts/security/oidcclient/mtls/AuthenticationTestSuite.java
Comment (michalvavrik) [reply security/oidc-client-mutual-tls/src/test/java/io/quarkus/ts/security/oidcclient/mtls/AuthenticationTestSuite.java]: Refactored, thanks!

Comment (michalvavrik) [reply security/oidc-client-mutual-tls/src/test/java/io/quarkus/ts/security/oidcclient/mtls/AuthenticationTestSuite.java]: Refactored, thanks!

File: security/oidc-client-mutual-tls/src/test/java/io/quarkus/ts/security/oidcclient/mtls/MutualTlsKeycloakService.java
Comment (michalvavrik) [reply security/oidc-client-mutual-tls/src/test/java/io/quarkus/ts/security/oidcclient/mtls/MutualTlsKeycloakService.java]: Fixed

File: security/oidc-client-mutual-tls/src/test/java/io/quarkus/ts/security/oidcclient/mtls/MutualTlsKeycloakService.java
Comment (michalvavrik) [reply security/oidc-client-mutual-tls/src/test/java/io/quarkus/ts/security/oidcclient/mtls/MutualTlsKeycloakService.java:10]: Shall do, let's merge this MR first and update it once I've moved `MutualTlsKeycloakService` to QuarkusTestFramework.

File: security/oidc-client-mutual-tls/src/test/resources/openshift-rh-sso-deployment-template.yml
Comment (michalvavrik) [reply security/oidc-client-mutual-tls/src/test/resources/openshift-rh-sso-deployment-template.yml:2]: In order to do that, I'll need to adjust at the very least `io.quarkus.test.services.containers.OpenShiftContainerManagedResource#replaceDeploymentContent` as `tls.crt` and `tls.key` should be configurable if we add ths template to Quarkus-test-framework. I will do this change together with the other issue.

**Comment by michalvavrik** [comment]
Hello @pjgg ,

thank you for the review. I added a reference to `README.md`, removed duplicate certificates and resolved issues mentioned in your comments. Regarding suggested QuarkusTestFramework enhancement, let's do it separately once this MR is merged and refactor the test suite afterwards.

BR,
Michal

**Pull request merged**
